src/encoded/types/biosample.py

```diff
@@ -205,6 +205,20 @@ def age_units(
         else:
             return model_organism_age_units
 
+    @calculated_property(define=True,
+                        schema={
+                            "title": "Disease term name",
+                            "description": "Ontology term describing the disease affecting the biosample.",
+                            "type": "string",
+                            "notSubmittable": True,
+                        })
+    def disease_term_name(self, request, registry, disease_term_id=None):
+        term_name = None
+        if disease_term_id is not None:
+            if disease_term_id in registry['ontology']:
+                term_name = registry['ontology'][disease_term_id]['name']
+        return term_name
```

Comment:
if you look on biosample_type, you will see that many of the calculated properties assume the term_id exists in the ontology, which ensures that the use of incorrect term_id will result in schema validation error - which is what prevent people from submitting incorrect or non-existing ids. then you don't need the audit checking for a match, please let me know if that makes sense?